packages/background/src/controllers/AccountTrackerController.ts

```diff
@@ -1156,4 +1186,29 @@ export class AccountTrackerController extends BaseController<AccountTrackerState
             return [];
         });
     }
+
+    public async getAccountNativeTokenBalanceForChain(
+        chainId: number
+    ): Promise<BigNumber | undefined> {
+        const selectedAddress =
+            this._preferencesController.getSelectedAddress();
+
+        const provider = this._networkController.getProviderForChainId(chainId);
+
+        if (provider === undefined) {
+            return undefined;
+        }
+        try {
+            const balances = await this._getAddressBalances(
```

Comment:
I understand why you're saying this but I don't think it is the best approach for this particular case, let me give a little bit of context.

When the user is performing a bridge, we need to check his/her "native" balance in the destination chain to check whether he/she has enough balance to cover fees and show proper warnings. As the native balances are updated once the user changes the network, we can't rely on the cached balance since it might have changed.